api/turing/api/ensemblers_api.go

```diff
@@ -0,0 +1,183 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	mlp "github.com/gojek/mlp/client"
+	"github.com/gojek/turing/api/turing/models"
+	"github.com/gojek/turing/api/turing/service"
+)
+
+type EnsemblersController struct {
+	BaseController
+}
+
+func (c EnsemblersController) ListEnsemblers(
+	r *http.Request,
+	vars RequestVars,
+	_ interface{},
+) *Response {
+	options := service.EnsemblersListOptions{}
+
+	if err := c.ParseVars(&options, vars); err != nil {
+		return BadRequest("unable to list ensemblers",
+			fmt.Sprintf("failed to parse query string: %s", err))
+	}
+
+	results, err := c.EnsemblersService.List(options)
+	if err != nil {
+		return InternalServerError("unable to list ensemblers", err.Error())
+	}
+
+	return Ok(results)
+}
+
+func (c EnsemblersController) GetEnsembler(
+	_ *http.Request,
+	vars RequestVars,
+	_ interface{},
+) *Response {
+	params := struct {
```

Comment:
Why not declare it as a named struct instead of defining it every HTTP call?

Reply:
I declared it as a named struct first, but that makes the validation error even less relevant (something like `Key: 'NamedStruct.EnsemblerID' Error:Field validation for 'NamedStruct.EnsemblerID' failed on the 'required' tag`). But I guess it doesn't matter that much, because errors returned by API are not translated into user-readable messages anyway.

Reply:
Done that here: https://github.com/gojek/turing/pull/55/commits/0f33f0e269f768fe9d7a008345c24d6e0385e0d2